src/huggingface_hub/utils/_xet.py

```diff
@@ -24,20 +24,28 @@ class XetConnectionInfo:
     endpoint: str
 
 
-def parse_xet_file_data_from_headers(headers: Dict[str, str]) -> Optional[XetFileData]:
+def parse_xet_file_data_from_headers_and_links(
+    headers: Dict[str, str], links: Dict[str, Dict[str, str]]
+) -> Optional[XetFileData]:
```

Comment:
```suggestion
def parse_xet_file_data_from_response(response: Response) -> Optional[XetFileData]:
```

Slight preference for parsing from the response directly since we are always calling it just after a request has been made. It makes the method implementation depend on `requests` but I don't think this is a problem (we are already bind quite a lot to `requests`).

Reply:
Switched this to use requests.Response as input, renamed function accordingly, and updates tests.